osu.Game/Screens/Edit/EditorBeatmap.cs

```diff
@@ -74,15 +75,27 @@ public EditorBeatmap(IBeatmap playableBeatmap)
         private void updateHitObject([CanBeNull] HitObject hitObject, bool silent)
         {
             scheduledUpdate?.Cancel();
-            scheduledUpdate = Scheduler.AddDelayed(() =>
+
+            if (hitObject != null)
+                pendingUpdates.Add(hitObject);
+
+            scheduledUpdate = Schedule(() =>
             {
```

Comment:
i guess using schedule is fine, although feels like `Cached` might be more appropriate.